examples/features/grid-spatial-filter.js

```diff
@@ -0,0 +1,292 @@
+                dataIndex: 'WARNCELLID',
+                flex: 1,
+                filter: {
+                    type: 'list'
+                }
+            }, {
+                text: 'String',
+                dataIndex: 'NAME',
+                flex: 2,
+                filter: {
+                    type: 'string'
+                }
+            }, {
+                text: 'Number',
+                dataIndex: 'WARNCELLID',
+                flex: 2,
+                filter: {
+                    type: 'number'
+                }
+            }, {
+                text: 'Date',
+                xtype: 'datecolumn',
+                formatter: 'date("Y-m-d")',
+                dataIndex: 'PROCESSTIME',
+                flex: 2,
+                filter: {
+                    type: 'date',
+                    dateFormat: 'Y-m-d'
+                }
+            }],
+            width: 600,
+            listeners: {
+                'filterchange': function(rec, filters) {
+                    activeFilters = filters;
+                    FeatureGridWithSpatialFilter.app.refreshFilters();
+                }
+            }
+        });
+        var mapComponent = Ext.create('GeoExt.component.Map', {
+            map: olMap
+        });
+        var mapPanel = Ext.create('Ext.panel.Panel', {
+            region: 'center',
+            height: 400,
+            layout: 'fit',
+            items: [mapComponent]
+        });
+        var description = Ext.create('Ext.panel.Panel', {
+            contentEl: 'description',
+            region: 'north',
+            title: 'Description',
+            height: 230,
+            border: false,
+            bodyPadding: 5,
+            autoScroll: true
+        });
+        Ext.create('Ext.Viewport', {
+            layout: 'border',
+            items: [description, mapPanel, gridWest]
+        });
+    },
+
+    /**
+     *
+     */
+    refreshFilters: function() {
+        var wmsFilter = GeoExt.util.OGCFilter.
+            getOgcWmsFilterFromExtJsFilter(activeFilters);
+        wmsLayer.getSource().updateParams({
+            filter: wmsFilter,
+            cacheBuster: Math.random()
+        });
+    },
+
+    onDrawEnd: function(evt) {
+        var feature = evt.feature;
+        var geometry = feature.getGeometry();
+        var typeName = 'THE_GEOM';
+        var spatialOperatorsCombo = Ext.ComponentQuery.
+            query('combobox[name="spatialOperatorsCombo"]')[0];
+        var operator = spatialOperatorsCombo.getValue();
+
+        // Create an instance of the spatial filter
+        var extFilter = GeoExt.util.OGCFilter.createSpatialFilter(
+            operator,
+            typeName,
+            geometry,
+            olMap.getView().getProjection().getCode()
+        );
+
+        if (extFilter) {
+            featStore.addFilter(extFilter);
+            this.refreshFilters();
+        }
+    },
+
+    createDrawInteraction: function(val, check) {
+        if (check) {
```

Comment:
What is exactly happening here?

Reply:
If the  value in the spatial operator combobox changed, by default a polygon should be drawn (if button is already toggled).

In b0245d6, I've enhanced the documentation and variable naming a bit to make this more clear.